lib/view/profiel/ProfielForm.php

```diff
@@ -228,12 +228,18 @@ public function __construct(Profiel $profiel) {
 		}
 
 		$fields[] = new Subkopje('<b>Einde vragenlijst</b><br /><br /><br /><br /><br />');
-		if ($admin OR LoginModel::mag('commissie:NovCie')) {
+		if (($admin OR LoginModel::mag('commissie:NovCie')) AND ($profiel->propertyMogelijk('novitiaat') || $inschrijven)) {
 			$fields[] = new CollapsableSubkopje('novcieForm', 'In te vullen door NovCie', true);
-			$fields[] = new RequiredTextareaField('novitiaat', $profiel->novitiaat, 'Wat verwacht Noviet van novitiaat?');
-			$fields[] = new RequiredSelectField('novietSoort', $profiel->novietSoort, 'Soort Noviet', array('noviet', 'nanoviet'));
-			$fields[] = new RequiredSelectField('matrixPlek', $profiel->matrixPlek, 'Matrix plek', array('voor', 'midden', 'achter'));
-			$fields[] = new RequiredSelectField('startkamp', $profiel->startkamp, 'Startkamp', array('ja', 'nee'));
+
+			$fields[] = new TextareaField('novitiaat', $profiel->novitiaat, 'Wat verwacht Noviet van novitiaat?');
+			end($fields)->required = $inschrijven;
```

Comment:
🤔 Niet zeker of ik hier een fan van ben. Wat er bijvoorbeeld in [LoginForm](https://github.com/csrdelft/csrdelft.nl/blob/master/lib/view/login/LoginForm.php#L26) wordt gedaan werkt ook erg goed en is niet gevoelig voor context, het is misschien iets minder mooi. Je kan nu de code breken door iets tussen L234 en L235 in te zetten.

Reply:
Ja, eens. Overigens vind ik classes als RequiredSelectField ook heel lelijk (ze veranderen alleen één property). Een extra argument toevoegen aan de constructor is ook weer zo wat. PHP heeft echt named parameters nodig.